File: app/src/main/java/com/bll/lnkstudy/dialog/ProgressDialog.java
package com.bll.lnkstudy.dialog;

import android.app.Activity;
import android.app.Dialog;
import android.content.Context;
import android.view.Gravity;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;

import com.bll.lnkstudy.R;
import com.bll.lnkstudy.ui.activity.HomeLeftActivity;


public class ProgressDialog {
    Context context;
    Dialog mDialog;
    int screenPos=0;//当前屏幕位置

    public ProgressDialog(Context context,int screenPos) {
        this.context = context;
        this.screenPos=screenPos;
        createDialog();
    }

    public void createDialog() {
        mDialog = new Dialog(context);
        mDialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        View view = View.inflate(context, R.layout.dialog_progress, null);
        mDialog.setContentView(view);
        mDialog.setCanceledOnTouchOutside(false);
        Window window = mDialog.getWindow();
        //要加上设置背景，否则dialog宽高设置无作用
        window.setBackgroundDrawableResource(android.R.color.transparent);
        WindowManager.LayoutParams layoutParams = window.getAttributes();
        //设置背景不变暗
        layoutParams.dimAmount=0f;
        WindowManager wm = (WindowManager) context.getApplicationContext().getSystemService(Context.WINDOW_SERVICE);
        int width = wm.getDefaultDisplay().getWidth();  //屏幕宽
        int height = wm.getDefaultDisplay().getHeight();  //屏幕高
        layoutParams.width = 250;
        layoutParams.height = 250;

        //全屏时 加载在a屏
        if (screenPos==3 ){
            layoutParams.gravity= Gravity.CENTER_VERTICAL|Gravity.LEFT;
            layoutParams.x=570;
        }

        //主页、全屏时 加载在b屏
        if (screenPos==3&&context instanceof HomeLeftActivity){
            layoutParams.gravity= Gravity.CENTER_VERTICAL|Gravity.RIGHT;
            layoutParams.x=570;
        }

        window.setAttributes(layoutParams);
    }



    public void show() {
        Activity activity= (Activity) context;
        if (activity!=null && !activity.isFinishing() && !activity.isDestroyed() &&
                mDialog != null && !mDialog.isShowing()) {
            mDialog.show();
        }
    }



    public void dismiss() {
        Activity activity= (Activity) context;
        if (activity!=null && !activity.isFinishing() && !activity.isDestroyed() &&
                mDialog != null && mDialog.isShowing()) {
            mDialog.dismiss();
        }
    }

}

File: app/src/main/java/com/bll/lnkstudy/dialog/ProgressDialog.kt
package com.bll.lnkstudy.dialog

import android.app.Activity
import android.app.Dialog
import android.content.Context
import android.view.Gravity
import android.widget.TextView
import com.bll.lnkstudy.Constants
import com.bll.lnkstudy.R
import com.bll.lnkstudy.utils.DP2PX

class ProgressDialog(var context: Context, val screenPos: Int,val type:Int) {
    var mDialog: Dialog? = null

    init {
        createDialog()
    }

    private fun createDialog() {
        mDialog = Dialog(context)
        mDialog!!.setContentView(R.layout.dialog_progress)
//        if (type!=1)
//            mDialog!!.setCanceledOnTouchOutside(false)
        val window = mDialog!!.window
        window!!.setBackgroundDrawableResource(android.R.color.transparent)
        val layoutParams = window.attributes
        if (screenPos == 1) {
            layoutParams.gravity = Gravity.CENTER_VERTICAL or Gravity.START
        } else {
            layoutParams.gravity = Gravity.CENTER_VERTICAL or Gravity.END
        }
        layoutParams.x=(Constants.WIDTH- DP2PX.dip2px(context,120f))/2
        window.attributes = layoutParams

        val tvName=mDialog?.findViewById<TextView>(R.id.tv_name)
        tvName?.text=if (type==0) "加载中..." else "网络连接中..."
    }

    fun setCanceledOutside(boolean: Boolean){
        mDialog?.setCanceledOnTouchOutside(boolean)
    }

    fun show() {
        val activity = context as Activity
        if (!activity.isFinishing && !activity.isDestroyed && mDialog != null && !mDialog!!.isShowing) {
            mDialog!!.show()
        }
    }

    fun dismiss() {
        val activity = context as Activity
        if (!activity.isFinishing && !activity.isDestroyed && mDialog != null && mDialog!!.isShowing) {
            mDialog!!.dismiss()
        }
    }
}